### PR TITLE
Generalized config expansion function.

### PR DESCRIPTION
**Pull request by jessemyers**
I've now written this three times, so I'm writing a generic function to expand a dict into a nested configuration.

### CODE REVIEW

File: microcosm/loaders.py [hunk 1]
```diff
@@ -97,30 +132,13 @@ def _load_from_environ(metadata, value_func=None):
...
-    for key, value in environ.items():
-        separator = "__" if "__" in key else "_"
-        key_parts = key.split(separator)
-        process_env_var(key_parts, value, config)
-    return config
+    return expand_config(
+        environ,
+        separator=lambda key: "__" if "__" in key else "_",
+        skip_to=len(prefix),
+        key_parts_filter=lambda key_parts: len(key_parts) > len(prefix) and key_parts[:len(prefix)] == prefix,
```
Comment (cdm-ium) [inline microcosm/loaders.py]: I'm a little confused here. Isn't the prefix a "key_part"? Wonder if the naming is off or if I'm missing something?

File: microcosm/tests/test_loaders.py [hunk 2]
```diff
@@ -50,6 +51,32 @@ def configfile(data):
...
+def test_expand_config():
+    """
+    Configuration expansion should work.
+
+    """
+    assert_that(
+        expand_config(
+            {
+                "prefix.foo": "bar",
+                "prefix.BAR.baz": lambda: "foo",
```
Comment (cdm-ium) [inline microcosm/tests/test_loaders.py]: Where does the desire for the default behavior to call values of the dictionary if they are callable come from?

It seems undesirable, for example, if they are handler mappings from microcosm pubsub and you only want to call them when you have a message to process. I like having the ability to special a value func, but not sure why calling is the default behavior.

**Comment by cdm-ium** [comment]
+1

File: microcosm/loaders.py (hunk 1 above)
Comment (jessemyers) [inline microcosm/loaders.py]: This logic is old, just got moved around.

The intention here is to only match environment variables that start with the right prefix so:
 - Don't match any keys that aren't long enough
 - Don't match any keys that don't have their first N parts equivalent to the prefix

File: microcosm/tests/test_loaders.py (hunk 2 above)
Comment (jessemyers) [inline microcosm/tests/test_loaders.py]: Fair point. I'll change it.

**Pull request closed without merging**